scripts/generate_node_indexes.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -eu
+
+MODULE_SRC_DIR=${1:?}
+
+find_module_files() {
+    for moduleFile in ${MODULE_FILES}; do
+        echo "${moduleFile}" | grep "^$1/[^/]*$" | sed "s,$1/,,g"
+    done
+}
+
+find_sub_dirs() {
+    for subDir in ${MODULE_DIRS}; do
+        echo "${subDir}" | grep "^$1/[^/]*$" | sed "s,$1/,,g"
+    done
+}
+
+pushd "${MODULE_SRC_DIR}" > /dev/null
+
+find . -type f \( -name '*.js' -o -name '*.d.ts' \) -exec grep -i 'GENERATED CODE -- NO SERVICES IN PROTO' {} \; -print -exec rm -f {} \;
```

Comment:
Not sure if `-delete` would be preferable to `-exec rm -f {} \;`

Reply:
New fangled :)